src/store/transaction/TransactionActions.js

```diff
@@ -49,42 +51,69 @@ export class TransactionActions {
   }
 
   @ThunkAction
-  prepareTransaction(id, from, to, amount) {
+  prepareTransaction(id, from, to, amount, platform) {
     return async dispatch => {
-      const toId = await this.accountResolver.getIdByAccountName(to);
-      const accountName = await this.accountResolver.getAccountNameById(toId);
-      if (accountName !== to) {
-        throw new Error('Can\'t resolve "to" account');
-      }
+      let transaction;
+
       const currentBalance = await this.privateSnaxProvider.getBalance(from);
-      const transaction = {
-        balance: currentBalance,
-        id,
-        from,
-        to: { platformId: toId, platformAccountName: to },
-        amount
-      };
-      dispatch(
-        this.setTransactionToSign(
-          currentBalance,
+
+      if (platform === 'twitter') {
+        const toId = await this.accountResolver.getIdByAccountName(to);
+        const accountName = await this.accountResolver.getAccountNameById(toId);
+
+        if (accountName !== to) {
+          throw new Error('Can\'t resolve "to" account');
+        }
+
+        transaction = {
+          balance: currentBalance,
           id,
           from,
-          { platformId: toId, platformAccountName: to },
-          amount
-        )
-      );
+          to: toId,
+          displayName: accountName,
+          amount,
+          platform,
+        };
+      } else if (platform === 'snax') {
+        if (from === to) {
+          throw new Error("Can't send transaction to the self");
+        }
+        transaction = {
+          balance: currentBalance,
+          id,
+          from,
+          to,
+          displayName: to,
+          amount,
+          platform,
+        };
+      } else {
```

Comment:
Platform may be not null, for instance Steemit but it will throw incorrect exception anyway

Reply:
I don't get it. After we will add the `Steemit` platform transaction platform will be `steemet`, am I right?

Reply:
It should throw something like `Platform ${platform} isnt supported` if platform != null

Reply:
Because if user passes some platform which is none of current supported platforms, he will receive strange error

Reply:
Got it. Make sense